### PR TITLE
vendor: update miekg/dns to v1.1.0

### PR DESCRIPTION
**Pull request by aanm**
Signed-off-by: André Martins <andre@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6339)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

hit https://github.com/cilium/cilium/issues/6341

**Review by raybejjani — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by ianvernon — APPROVED** [review]

File: Gopkg.toml [hunk 1]
```diff
@@ -405,7 +405,7 @@ required = [
   # main-usage = "only in integration tests"
   # on-revision = ""
 
-[[constraint]]
+[[override]]
```
Comment (jrajahalme) [inline Gopkg.toml:408]: Is this change related to the miekg/dns version change? Why?

**Review by jrajahalme — CHANGES_REQUESTED** [review]
Please provide a short motivation for the change in the commit message.

Also, there is one change that seems unrelated.

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
@jrajahalme let me know if the changes are good now.

**Review by tgraf — APPROVED** [review]

**Pull request merged**